### PR TITLE
[SofaOpenglVisual] Pluginize.

### PR DESCRIPTION
**Pull request by damienmarchal**
______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: applications/plugins/SofaMiscCollision/CMakeLists.txt [hunk 1]
```diff
@@ -71,10 +71,6 @@ add_library(${PROJECT_NAME} SHARED ${HEADER_FILES} ${SOURCE_FILES} ${EXTRA_FILES
 target_link_libraries(${PROJECT_NAME} PUBLIC SofaBaseTopology SofaGeneralMeshCollision SofaGeneralDeformable SofaConstraint SofaExplicitOdeSolver SofaGeneralExplicitOdeSolver SofaImplicitOdeSolver SofaBaseCollision)
 target_link_libraries(${PROJECT_NAME} PUBLIC SofaEigen2Solver)
 
-if(NOT SOFA_NO_OPENGL)
```
Comment (fredroy) [inline applications/plugins/SofaMiscCollision/CMakeLists.txt:74]: good job  by removing the dep (and replacing oglmodel by visualmodelimpl) !

File: applications/plugins/Geomagic/CMakeLists.txt [hunk 2]
```diff
@@ -28,7 +29,7 @@ include_directories( ${OPENHAPTICS_INCLUDE_DIR})
 add_library(${PROJECT_NAME} SHARED ${HEADER_FILES} ${SOURCE_FILES} ${SCENES_FILES} ${README_FILES})
 set_target_properties(${PROJECT_NAME} PROPERTIES COMPILE_FLAGS "-DPLUGIN_DATA_DIR=\"${CMAKE_CURRENT_SOURCE_DIR}/\"")
 set_target_properties(${PROJECT_NAME} PROPERTIES COMPILE_FLAGS "-DSOFA_BUILD_GEOMAGIC")
-target_link_libraries(${PROJECT_NAME} SofaHelper SofaUserInteraction SofaHaptics ${OPENHAPTICS_LIBRARIES})
+target_link_libraries(${PROJECT_NAME} SofaHelper SofaUserInteraction SofaHaptics SofaOpenglVisual ${OPENHAPTICS_LIBRARIES})
```
Comment (fredroy) [inline applications/plugins/Geomagic/CMakeLists.txt:32]: You made SofaOpenGLVisual required for the Geomagic plugin, intended ?

Comment (damienmarchal) [reply applications/plugins/Geomagic/CMakeLists.txt:32]: Sadely Geomagic & other uses OglModel in their cpp code. 
So it just make obvious that there is a dependency.

File: applications/plugins/Geomagic/CMakeLists.txt [hunk 3]
```diff
@@ -28,7 +29,7 @@ include_directories( ${OPENHAPTICS_INCLUDE_DIR})
...
 include_directories("${CMAKE_CURRENT_SOURCE_DIR}/..")
 
 if(NOT ${SOFA_NO_OPENGL})
```
Comment (fredroy) [inline applications/plugins/Geomagic/CMakeLists.txt]: so you can delete those lines ;)

**Comment by damienmarchal** [comment]
[ci-build][with-scene-tests][with-regression-tests]

**Pull request merged**